vital/types/detected_object_set.h

```diff
@@ -28,202 +28,168 @@ class detected_object_set;
-   * problem, clone() this set before shifting.
-   *
-   * @param col_shift Column  (a.k.a. x, i, width) translation factor
-   * @param row_shift Row (a.k.a. y, j, height) translation factor
-   */
-  void
-  VITAL_DEPRECATED
-  shift( double col_shift, double row_shift );
-
-  /**
-   * @brief Get attributes set.
-   *
-   * This method returns a pointer to the attribute set that is
-   * attached to this object. It is possible that the pointer is NULL,
-   * so check before using it.
-   *
-   * @return Pointer to attribute set or NULL
-   */
+  ///@}
+
+  /// Select detections based on confidence value.
+  ///
+  /// This method returns a vector of detections ordered by confidence value,
+  /// from high to low. If the optional \p threshold is specified, then all
+  /// detections from the set that are less than the threshold are not in the
+  /// selected set. Note that the selected set may be empty.
+  ///
+  /// The returned vector refers to the actual detections in the set, so if you
+  /// make changes to the selected set, you are also changing the object in the
+  /// set. If you want a clean set of detections, call clone() first.
+  ///
+  /// \param threshold
+  ///   Select all detections with confidence not less than this value. If this
+  ///   parameter is omitted, all detections are selected.
+  ///
+  /// \return List of detections.
+  detected_object_set_sptr select(
+    double threshold = detected_object_type::INVALID_SCORE ) const;
+
+  /// Select detections based on class name.
+  ///
+  /// This method returns a vector of detections that have the specified
+  /// \p class_name. These detections are ordered by descending score for the
+  /// name. Note that the selected set may be empty.
+  ///
+  /// The returned vector refers to the actual detections in the set, so if you
+  /// make changes to the selected set, you are also changing the object in the
+  /// set. If you want a clean set of detections, call clone() first.
+  ///
+  /// \param class_name Class name of detections to be selected.
+  /// \param threshold
+  ///   Select all detections with confidence not less than this value. If this
+  ///   parameter is omitted, all detections are selected.
+  ///
+  /// \return List of detections.
+  detected_object_set_sptr select(
+    const std::string& class_name,
+    double threshold = detected_object_type::INVALID_SCORE ) const;
+
+  /// Scale all detection locations by some scale factor.
+  ///
+  /// This method changes the bounding boxes within all stored detections by
+  /// scaling them by the specified scale factor.
+  ///
+  /// \note
+  ///   Detections in this set can be shared by multiple sets, so scaling the
+  ///   detections in this set will also scale the detection in other sets that
+  ///   share the detections. To avoid this, clone() the set before shifting.
+  ///
+  /// \param scale Scale factor to be applied.
+  ///
+  /// \deprecated
+  [[deprecated]]
```

Comment:
`[[deprecated]]` is C++14, but I believe KWIVER claims to support C++11.  Should we continue to use `VITAL_DEPRECATED` for now?

My vote would be to not introduce this change as part of this branch, but to decide if we should move to C++14, and if so, make a new branch that replaces `VITAL_DEPRECATED` with `[[deprecated]]` globally.

Reply:
I'll change that back presently. The main reason I changed it was because it "spuriously" shows up in the generated documentation, but there's an alternate fix for that. (Also it should be *before* the type so that we can use a C++11-style attribute rather than `__attribute__`.)